examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/paywall/PaywallScreenViewModel.kt

```diff
@@ -35,6 +39,42 @@ class PaywallScreenViewModelImpl(
         updateOffering()
     }
 
+    override fun onPurchaseCompleted(customerInfo: CustomerInfo, storeTransaction: StoreTransaction) {
+        val value = _state.value
+        if (value is PaywallScreenState.Loaded) {
+            _state.update {
+                value.copy(
+                    displayCompletedPurchaseMessage = true,
+                    displayErrorPurchasingMessage = false,
+                )
+            }
+        }
+    }
+
+    override fun onPurchaseError(error: PurchasesError) {
+        val value = _state.value
+        if (value is PaywallScreenState.Loaded) {
+            _state.update {
+                value.copy(
+                    displayCompletedPurchaseMessage = false,
+                    displayErrorPurchasingMessage = true,
+                )
+            }
+        }
+    }
+
+    override fun onDismissed() {
```

Comment:
Hmm this is using the `PaywallViewListener` method... My original idea for this was that, once we implement helper presentation systems for the paywall (whether it's dialogs or bottom sheets), this could be called once it was dismissed.

In this case, what we are dismissing is not related to the `RevenueCatUI` library, so I would create a new method in the view model called `onDialogDismissed` or something like that.

Reply:
ohhh got it, makes sense